kong/utils.go

```diff
@@ -103,3 +113,51 @@ func HTTPClientWithHeaders(client *http.Client,
 	}
 	return res
 }
+
+func cleanSemVer(v string) (semver.Version, error) {
+	// fix enterprise edition semver adding patch number
+	// fix enterprise edition version with dash
+	// fix bad version formats like 0.13.0preview1
+	re := regexp.MustCompile(`(\d+\.\d+)(?:[\.-](\d+))?(?:\-?(.+)$|$)`)
+	m := re.FindStringSubmatch(v)
+	if len(m) != versionParts {
+		return semver.Version{}, fmt.Errorf("Unknown Kong version : '%v'", v)
+	}
+	if m[2] == "" {
+		m[2] = "0"
+	}
+	if m[3] != "" {
+		m[3] = "-" + strings.Replace(m[3], "enterprise-edition", "enterprise", 1)
+		m[3] = strings.Replace(m[3], ".", "", -1)
+	}
+	v = fmt.Sprintf("%s.%s%s", m[1], m[2], m[3])
+	return semver.Make(v)
+}
+
+func getKong(root map[string]interface{}) (*Kong, error) {
+	version := root["version"].(string)
+	configuration := root["configuration"].(map[string]interface{})
+	semVer, err := cleanSemVer(version)
```

Comment:
Let's not do this here since it is a lossy conversion.
The user of the library can interpret the version if needed.

Reply:
How about providing both the cleaned semver and the original string with all our extra content? Elsewhere we're likely going to be performing semver comparisons for the most part, and I think it makes sense to have a standard conversion rather than each user implementing their own `cleanSemVer()` variant.

Reply:
I think a better approach would be:
```go
info, err := client.Root() // most powerful
version := kong.versionFromInfo(info) // package-level global function to fetch version from the info map
semanticVersion := utils.ParseSemanticVersion(version) // package-level function
```

This makes the code cleaner and keeps each function responsible for one and only one thing.